src/network/core.rs

```diff
@@ -232,6 +232,180 @@ impl Core {
         }
     }
 
+    pub fn macvlan_per_podman_network(
+        per_network_opts: &types::PerNetworkOptions,
+        network: &types::Network,
+        netns: &str,
+    ) -> Result<types::StatusBlock, std::io::Error> {
+        //  StatusBlock response
+        //  StatusBlock response
+        let mut response = types::StatusBlock {
+            interfaces: Some(HashMap::new()),
+        };
+        // Does config have a macvlan mode ? I think not
+        // Important !! Hardcode MACVLAN_MODE to bridge
+        let macvlan_mode: u32 = 4u32;
+        // get master interface name
+        let mut master_ifname: String = network.network_interface.as_ref().unwrap().to_owned();
```

Comment:
You cannot use unwrap here. This can be empty. Your code below must handle this if it is None.

Reply:
Yes just checking for `empty` string in line after this and getting default `Route` interface if this is empty.

Reply:
ackd. I'll rectify this.